app/models/mixins/miq_policy_mixin.rb

```diff
@@ -29,7 +29,6 @@ def remove_policy(policy)
   def get_policies
     policy_tags
       .map { |t| t.split("/").first(2) }
-      .group_by(&:first)
       .select { |klass, _ids| ["miq_policy", "miq_policy_set"].include?(klass) }
       .flat_map { |klass, ids| klass.camelize.constantize.where(:id => ids).to_a }
   end
```

Comment:
This change was introduced late in the game.

I think the `group_by` is correct for the most part.
That is what allowed us to consolidate the sql queries from N to 0..2

I think the value in `ids` is `[[klass, id],[klass, id], [klass, id]`
The code looks to assume `ids` is `[id, id, id]`
so I introduced the `map(&:last)

That is the part to check out.

```diff
+      .group_by(&:first)
       .select { |klass, _ids| ["miq_policy", "miq_policy_set"].include?(klass) }
-      .flat_map { |klass, ids| klass.camelize.constantize.where(:id => ids).to_a }
+      .flat_map { |klass, ids| klass.camelize.constantize.where(:id => ids.map(&:last)).to_a }
```

Reply:
thanks, you're right